TestSyncTrigger time out

```
==================
WARNING: DATA RACE
Write by goroutine 39:
  github.com/coreos/etcd/etcdserver.(*raftNode).run()
      github.com/coreos/etcd/etcdserver/_test/_obj_test/raft.go:89 +0x9d
Previous read by goroutine 34:
  github.com/coreos/etcd/etcdserver.func·003()
      github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:474 +0x7e
  github.com/coreos/etcd/etcdserver.(*EtcdServer).run()
      github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:558 +0x17e5
Goroutine 39 (running) created at:
  github.com/coreos/etcd/etcdserver.(*EtcdServer).run()
      github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:471 +0x351
Goroutine 34 (running) created at:
  github.com/coreos/etcd/etcdserver.(*EtcdServer).start()
      github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:388 +0x5c7
  github.com/coreos/etcd/etcdserver.TestSyncTrigger()
      /home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server_test.go:666 +0x603
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:447 +0x133
==================
panic: test timed out after 3m0s
goroutine 249 [running]:
testing.func·008()
	/usr/local/go/src/testing/testing.go:681 +0x168
created by time.goFunc
	/usr/local/go/src/time/sleep.go:129 +0x59
goroutine 1 [chan receive]:
testing.RunTests(0xaeff80, 0x1607140, 0x37, 0x37, 0xc20809c201)
	/usr/local/go/src/testing/testing.go:556 +0xdc3
testing.(*M).Run(0xc2080566e0, 0x1611d60)
	/usr/local/go/src/testing/testing.go:485 +0xe1
main.main()
	github.com/coreos/etcd/etcdserver/_test/_testmain.go:222 +0x382
goroutine 200 [chan send (nil chan)]:
github.com/coreos/etcd/etcdserver.func·003()
	github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:474 +0xa9
github.com/coreos/etcd/etcdserver.(*EtcdServer).run(0xc208160480)
	github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:558 +0x17e6
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).start
	github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:388 +0x5c7
goroutine 199 [chan receive]:
github.com/coreos/etcd/etcdserver.(*EtcdServer).Stop(0xc208160480)
	github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:573 +0x188
runtime.Goexit()
	/usr/local/go/src/runtime/panic.go:288 +0xf6
testing.(*common).FailNow(0xc2080807e0)
	/usr/local/go/src/testing/testing.go:329 +0x5e
testing.(*common).Fatalf(0xc2080807e0, 0xa579b0, 0x18, 0xc2081b1dc0, 0x1, 0x1)
	/usr/local/go/src/testing/testing.go:370 +0x98
github.com/coreos/etcd/etcdserver.TestSyncTrigger(0xc2080807e0)
	/home/travis/gopath/src/github.com/yichengq/etcd/gopath/src/github.com/coreos/etcd/etcdserver/server_test.go:680 +0x88a
testing.tRunner(0xc2080807e0, 0x1607518)
	/usr/local/go/src/testing/testing.go:447 +0x134
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:555 +0xd4e
goroutine 257 [select]:
github.com/coreos/etcd/etcdserver.(*raftNode).run(0xc208160490)
	github.com/coreos/etcd/etcdserver/_test/_obj_test/raft.go:127 +0xf0b
created by github.com/coreos/etcd/etcdserver.(*EtcdServer).run
	github.com/coreos/etcd/etcdserver/_test/_obj_test/server.go:471 +0x351
FAIL	github.com/coreos/etcd/etcdserver	180.047s

```

--- a/etcdserver/raft.go
+++ b/etcdserver/raft.go
@@ -109,9 +109,6 @@ type raftNode struct {
 }
 
 func (r *raftNode) run() {
-	r.stopped = make(chan struct{})
-	r.done = make(chan struct{})
-
 	var syncC <-chan time.Time
 
 	defer r.stop()

--- a/etcdserver/raft_test.go
+++ b/etcdserver/raft_test.go
@@ -152,6 +152,8 @@ func TestStopRaftWhenWaitingForApplyDone(t *testing.T) {
 		storage:     &storageRecorder{},
 		raftStorage: raft.NewMemoryStorage(),
 		transport:   &nopTransporter{},
+		stopped:     make(chan struct{}),
+		done:        make(chan struct{}),
 	}
 	r.s = &EtcdServer{r: r}
 	go r.run()

--- a/etcdserver/server.go
+++ b/etcdserver/server.go
@@ -416,6 +416,8 @@ func (s *EtcdServer) run() {
 	// TODO: get rid of the raft initialization in etcd server
 	s.r.s = s
 	s.r.applyc = make(chan apply)
+	s.r.stopped = make(chan struct{})
+	s.r.done = make(chan struct{})
 	go s.r.run()
 	defer func() {
 		s.r.stopped <- struct{}{}

